### PR TITLE
ci: Changes xcode version to latest stable

### PR DESCRIPTION
**Pull request by NeedleInAJayStack**
This changes the MacOS CI from using the `latest` XCode to `latest-stable` to avoid going to v16, which comes with Swift 6.0.0. This resolves the issues observed here: https://github.com/GraphQLSwift/Graphiti/pull/137#issuecomment-2177314720

It also adds Swift 5.10 to the Linux test matrix

### CODE REVIEW

**Comment by abidon** [comment]
Hi @paulofaria,

Do you have any news regarding the review of this PR ?

PR #137 is blocked until this one is approved.

**Review by paulofaria — APPROVED** [review]

**Comment by paulofaria** [comment]
Sorry! Completely missed this PR. LGTM!

**Pull request merged**
